### DISCUSSION TITLE
v5.1

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#12024] [General]
## Release Notes Highlights

* Required: **.NET 8.0 Desktop Runtime [v8.0.10](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-8.0.10-windows-x64-installer)** or later
* Recommended: Git 2.46.0 or later
* Several bugfixes including for diff display in case `diff.external` is configured for Git
* Allow `^` and `..` in manual branch filters
* Improve in-line diff highlight, add anchor markers indicating where text was added / removed
* Add `Output` tab with the output from interactively run Git commands
* Performance improvements
* UI and usability improvements

## What's Changed
* fix: bold attributes applies to both fore/back by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11849
* feat: Recommend Git 2.46.0 by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11853
* feat: GitCache: Add current directory to key by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11850
* fix: try-catch for FileSystemWatcher by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11855
* fix: RunCommand() should not throw when exit is checked by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11871
* Update README.md to v5.0 by @RussKie in https://github.com/gitextensions/gitextensions/pull/11889
* fix: merge adjacent textmarkers by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11887
* Fix NRE in scripts if `arguments` unset by @RussKie in https://github.com/gitextensions/gitextensions/pull/11893
* fix: LineNumber presentation for Git word-diff by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11888
* Correctly escape paths by @RussKie in https://github.com/gitextensions/gitextensions/pull/11900
* Add navigation commands "Go to first / last parent commit" by @mstv in https://github.com/gitextensions/gitextensions/pull/11884
* Move NoWarn=NU1903 to the Wix PackageReference by @ericstj in https://github.com/gitextensions/gitextensions/pull/11911
* FileStatusList: Reuse context menu separators by @mstv in https://github.com/gitextensions/gitextensions/pull/11916
* FormCommit: Disable ResetSoft if command not applicable by @mstv in https://github.com/gitextensions/gitextensions/pull/11917
* fix: support Git < 2.26 with bright colors by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11915
* Make DropDownWidth calculation Dpi aware and account for vertical scrollbar by @mdonatas in https://github.com/gitextensions/gitextensions/pull/11864
* Adjust ToolStripComboBox size based on DPI scaling by @mdonatas in https://github.com/gitextensions/gitextensions/pull/11865
* feat(settings): deeper checks for GitExtensions install directory by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11913
* chore: Update wix to v3.14.1 (security update) by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11912
* fix: Always present git inserted as Header by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11922
* fix: Merge git-diff textmarkers separated by newlines by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11921
* Use "real" GitUICommands instance by @RussKie in https://github.com/gitextensions/gitextensions/pull/11927
* fix: Allow ^and .. in manual branch filters by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11923
* fix!: View colors in git-blob diff files by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11930
* Use "-D" instead of "-d" when deleting unmerged branches by @georg138 in https://github.com/gitextensions/gitextensions/pull/11937
* Diff View: Match similar lines in changed blocks by @mstv in https://github.com/gitextensions/gitextensions/pull/11851
* Expose git-grep in settings and context menu by @RussKie in https://github.com/gitextensions/gitextensions/pull/11858
* fix(show git-grep): Sync and use actual visibility by @mstv in https://github.com/gitextensions/gitextensions/pull/11948
* fix: Align difftastic colors by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11946
* fix(FormGoToCommit): Improve performance by @mstv in https://github.com/gitextensions/gitextensions/pull/11952
* Fix FileViewer's encoding bug when showing unstaged files by @mdonatas in https://github.com/gitextensions/gitextensions/pull/11960
* fix: Reset current ansi terminal color by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11962
* fix(ReportDubiousOwnership): Support normal paths, too by @mstv in https://github.com/gitextensions/gitextensions/pull/11965
* fix: commit gpgSign Git defaults by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11963
* fix(settings): Ignore non-boolean advanced git config settings by @mstv in https://github.com/gitextensions/gitextensions/pull/11959
* Improve in-line diff highlight and add anchor markers by @mstv in https://github.com/gitextensions/gitextensions/pull/11862
* fix(FormRenameBranch): Run git interactively by @mstv in https://github.com/gitextensions/gitextensions/pull/11987
* fix(push): fix tracking reference not well selected by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11954
* feat: Avoid coloring moved lines by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11947
* Improve unit tests code by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11993
* fix: Handle dubious repo ownership, too, if detected by GitStatusMonitor by @mstv in https://github.com/gitextensions/gitextensions/pull/11992
* fix: Restrict settings to "global" when not in a repository by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11955
* feat: Cache the diff highlighting by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11981
* fix(in-line diff): Support non-reversed git coloring by @mstv in https://github.com/gitextensions/gitextensions/pull/11994
* Singed for contributing  by @chkoddi in https://github.com/gitextensions/gitextensions/pull/11995
* fix(Push): fix false positive detection of "detached head" state by @pmiossec in https://github.com/gitextensions/gitextensions/pull/11999
* Add control with last interactive Git outputs and traces by @mstv in https://github.com/gitextensions/gitextensions/pull/10307
* fix(difftastic): Adjust endmarkers by @gerhardol in https://github.com/gitextensions/gitextensions/pull/11983
* fix(FileStatusList): Set `Text` for accessibility by @mstv in https://github.com/gitextensions/gitextensions/pull/12004
* fix(FormDiff): Avoid "A task was canceled." by @mstv in https://github.com/gitextensions/gitextensions/pull/12012
* fix(FormCommit): Restore file selection by @mstv in https://github.com/gitextensions/gitextensions/pull/12013
* fix(FilterToolBar): Correctly resize branch dropdown by @mstv in https://github.com/gitextensions/gitextensions/pull/12014
* fix(git diff): Suppress `diff.external` by @mstv in https://github.com/gitextensions/gitextensions/pull/12007
* fix(File History): Do not change capitalization of filename by @mstv in https://github.com/gitextensions/gitextensions/pull/12016
* fix: Add missing defaults to git attributes parser by @gerhardol in https://github.com/gitextensions/gitextensions/pull/12018

## New Contributors
* @ericstj made their first contribution in https://github.com/gitextensions/gitextensions/pull/11911
* @georg138 made their first contribution in https://github.com/gitextensions/gitextensions/pull/11937
* @chkoddi made their first contribution in https://github.com/gitextensions/gitextensions/pull/11995

**Full Changelog**: https://github.com/gitextensions/gitextensions/compare/v5.0...v5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v5.1'>v5.1</a>.</em>

**Comment by chirontt** [comment]
A manual build of this v5.1 version for Windows on Arm64 is available in my fork: https://github.com/chirontt/gitextensions/releases/tag/v5.1-arm64

**Comment by namtab00** [comment]
I've noticed 5.1 is already available on winget.

It got there via an [automated PR](https://github.com/microsoft/winget-pkgs/pull/189530) done with [komac](https://github.com/russellbanks/Komac).

For curiosity's sake, how was that setup / who has their eyes on that?

I'm asking because I, of my own initiative, manually did [the previous version's winget packaging](https://github.com/gitextensions/gitextensions/discussions/11599#discussioncomment-10953621), which was clearly a bit of a useless rollercoaster if someone else has their grip on the process.
